tools/AzureRM.BootStrapper/AzureRM.Bootstrapper.psm1

```diff
@@ -66,38 +146,478 @@ function Get-AzProfile
   PROCESS
   {
     $Update = $PSBoundParameters.Update
-    # If Update is present, download ProfileMap from Storage blob endpoint
+    # If Update is present, download ProfileMap from online source
     if ($Update.IsPresent)
     {
       return (Get-AzureProfileMap)
     }
 
     # Check the cache
-    $ProfileCache = Get-ProfileCachePath
-    if(Test-Path $ProfileCache)
+    $LatestProfileMapPath = Get-LatestProfileMapPath
```

Comment:
Added $LatestProfileMapPath as a script variable which will be calculated once and updated only when a new profile map is downloaded from online.